udata_hydra/analysis/csv.py

```diff
@@ -129,7 +129,14 @@ async def analyse_csv(
     # Check if the resource is in the exceptions table
     # If it is, get the table_indexes to use them later
     exception: Record | None = await ResourceException.get_by_resource_id(resource_id)
-    table_indexes: dict | None = json.loads(exception["table_indexes"]) if exception else None
+    table_indexes: dict | None = None
+    if exception:
+        try:
+            table_indexes: dict | None = json.loads(exception["table_indexes"])
+        except TypeError:
+            raise TypeError(
+                f"table_indexes should be a JSON object, got {exception.get('table_indexes')}"
+            )
```

Comment:
Could you do it like this?
```suggestion
            table_indexes: dict | None = json.loads(exception.get("table_indexes", "{}") or "{}")
```
exception["table_indexes"] is actually a *JSON string* and not a *JSON object*

Reply:
I think we want the var table_indexes to actually be None if it's empty in the DB, so that it doesn't even go through the index creation logic, but we can do:

```python
if exception and exception.get("table_indexes"):
    print("something in table indexes")
    table_indexes = json.loads(exception.get("table_indexes", "{}"))
```

Reply:
Changed. Will merge in a bit if that's OK